packages/extension-chakra-storefront/src/components/field/index.jsx

```diff
@@ -64,14 +54,24 @@ const Field = ({
                     typeof inputProps === 'function' ? inputProps({value, onChange}) : inputProps
 
                 return (
-                    <FormControl id={name} isInvalid={error}>
+                    <Field.Root id={name} invalid={!!error}>
                         {!['checkbox', 'radio', 'hidden'].includes(type) &&
-                            (formLabel || <FormLabel>{label}</FormLabel>)}
+                            (formLabel || <Field.Label>{label}</Field.Label>)}
 
-                        <InputGroup>
-                            {['text', 'password', 'email', 'phone', 'tel', 'number'].includes(
-                                type
-                            ) && (
+                        {['text', 'password', 'email', 'phone', 'tel', 'number'].includes(type) && (
+                            <InputGroup
+                                endElement={
+                                    type === 'password' ? (
+                                        <IconButton
+                                            variant="ghosted"
+                                            aria-label={passwordIconLabel}
+                                            onClick={() => setHidePassword(!hidePassword)}
+                                        >
+                                            <PasswordIcon color="gray.500" boxSize={6} />
+                                        </IconButton>
+                                    ) : undefined
+                                }
+                            >
```

Comment:
InputGroup (and no more InputRightElement) is not mentioned in the migration txt either. This one actually triggers a cryptic error message, which said something about how InputGroup accepts only 1 child as the children.